emcc.py

```diff
@@ -188,22 +188,29 @@ def base64_encode(b):
 
 @unique
 class OFormat(Enum):
+  # Output a relocatable object file.  We use this
+  # today for `-r` and `-shared`.
+  OBJECT = auto()
   WASM = auto()
   JS = auto()
   MJS = auto()
   HTML = auto()
   BARE = auto()
 
 
+@unique
+class Mode(Enum):
+  PREPROCESS_ONLY = auto()
+  PCH = auto()
+  COMPILE_ONLY = auto()
+  POST_LINK_ONLY = auto()
```

Comment:
does this include ld? Or if not, is there a mode for ld + post-link? (I'd expect that, along with compile-only to be one of the most common?)

Reply:
I think that is the default `COMPILE_AND_LINK` mode (which will not actually do any compiling unless you pass source files).